source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,296 @@
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "StartConditionsPanel.h"
+
+#include "Color.h"
+#include "Command.h"
+#include "ConversationPanel.h"
+#include "DataFile.h"
+#include "Dialog.h"
+#include "Files.h"
+#include "FillShader.h"
+#include "GameData.h"
+#include "GameWindow.h"
+#include "Information.h"
+#include "Interface.h"
+#include "LoadPanel.h"
+#include "MainPanel.h"
+#include "Messages.h"
+#include "PlayerInfo.h"
+#include "Planet.h"
+#include "Preferences.h"
+#include "Rectangle.h"
+#include "ShipyardPanel.h"
+#include "StarField.h"
+#include "System.h"
+#include "text/alignment.hpp"
+#include "text/DisplayText.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "text/Format.h"
+#include "text/layout.hpp"
+#include "text/truncate.hpp"
+#include "UI.h"
+
+#include <algorithm>
+
+using namespace std;
+
+namespace {
+	const int PAGE_INTERVAL = 10;
+}
+
+StartConditionsPanel::StartConditionsPanel(PlayerInfo &player, UI &gamePanels, Panel *parent)
+	: player(player), gamePanels(gamePanels), descriptionWrappedText(FontSet::Get(14)), parent(parent)
+{
+	const Interface *startConditionsMenu = GameData::Interfaces().Find("start conditions menu");
+	
+	if(startConditionsMenu)	
+	{
+		// Ideally, we want the content of the boxes to be drawn in Interface.cpp.
+		// However, we'd need a way to specify arbitrarily extensible lists there.
+		// We also would need to a way to specify truncation for such a list.
+		// Such a list would also have to be scrollable.
+		descriptionBox = startConditionsMenu->GetBox("start description");
+		entryBox = startConditionsMenu->GetBox("start entry");
+		entryListBox = startConditionsMenu->GetBox("start entry list");
+		entryInternalBox = startConditionsMenu->GetBox("start entry internal");	
+	}
+	
+	if(!GameData::StartOptions().empty())
+	{
+		// By default, select the last defined start conditions
+		// (which are usually the ones defined by a plugin).
+		chosenStart = &GameData::StartOptions().back();
+		chosenStartIterator = GameData::StartOptions().end() - 1;
+		listScroll = (chosenStartIterator - GameData::StartOptions().begin()) * entryBox.Height();
+	}
+	
+	const Rectangle firstRectangle = Rectangle::FromCorner(entryListBox.TopLeft(), entryBox.Dimensions());
+	startConditionsClickZones.reserve(GameData::StartOptions().size());
+	for(size_t i = 0; i < GameData::StartOptions().size(); ++i)
+		startConditionsClickZones.emplace_back(firstRectangle + Point(0, i * entryBox.Height()), GameData::StartOptions().begin() + i);
+	
+	descriptionWrappedText.SetAlignment(Alignment::LEFT);
```

Comment:
Why do you use Alignment::LEFT instead of Alignment::JUSTIFIED?

Reply:
I've removed the left alignment. It's possible some of the justified alignments made @FranchuFranchu 's chosen description text look odd. With justification, the "backstory" description looks good:
![image](https://user-images.githubusercontent.com/20871346/108631229-64990680-742e-11eb-82b9-0e201144f50e.png)